pkg/storage/snapshot/restore.go

```diff
@@ -143,6 +161,15 @@ func (ctrl *VMRestoreController) updateVMRestore(vmRestoreIn *snapshotv1.Virtual
 		return 0, nil
 	}
 
+	ready, err := target.Ready()
+	if err != nil {
+		logger.Reason(err).Error("Error checking target ready")
+		return 0, ctrl.doUpdateError(vmRestoreIn, err)
+	}
+	if !ready {
+		return 0, ctrl.handleVMRestoreTargetNotReady(vmRestoreOut, target)
+	}
+
```

Comment:
Should this block be moved until after owner refs added?

Reply:
Let's consider this: What if user chooses eventually readiness and the VM is deleted before the restore starts? The VM gets deleted but the restore does not since no owner ref. I think what will happen is that the restore will create a new VM. Is that what we want?

Reply:
May have to make sure that `Update()` is called on the restore as well

Reply:
what do you mean?

Reply:
Just have to make sure that the owner refs get persisted

Reply:
it is being done since we have the check of not equal and the update before the ready check so its ok.